engine/src/main/java/org/camunda/bpm/engine/impl/pvm/runtime/operation/PvmAtomicOperationDeleteCascadeFireActivityEnd.java

```diff
@@ -72,7 +72,8 @@ protected void eventNotificationsCompleted(PvmExecutionImpl execution) {
 
     } else {
       if (execution.isScope()) {
-        if(execution instanceof ExecutionEntity && !execution.isProcessInstanceExecution() && execution.isCanceled()) {
+        if(isOutputMappingSkippableForActivity(activity) && execution.isCanceled()) {
```

Comment:
I know it's a micro-optimization 😁, but the `execution.isCanceled()` is quite cheap and probably very selective (doesn't occur that often) - putting it first, we don't have to check the activity type in most cases I would imagine?
Feel free to ignore it, it just caught my eye 😃

Reply:
Sure let's do it :)